src/app/core/services/marios.service.ts

```diff
@@ -0,0 +1,85 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, ReplaySubject, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { Marios } from '../models/marios.model';
+import { MariosList } from '../models/marios-list.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MariosService {
+  private baseUrl = '/api/marios';
+  private mariosData: Marios[] = [];
+  private mariosList$ = new ReplaySubject<Marios[]>(1);
+  private destroy$ = new Subject<void>();
+
+  constructor(private http: HttpClient) {}
+
+  get mariosList(): Observable<Marios[]> {
+    if (this.mariosData.length === 0) {
+      this.fetchMariosList();
+    }
+    return this.mariosList$.asObservable();
+  }
+
+  private fetchMariosList(): void {
+    this.http
+      .get<Marios[]>(this.baseUrl)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((data) => {
+        this.mariosData = [...data];
+        this.mariosList$.next(this.mariosData);
+      });
+  }
+
+  getMariosById(mariosId: string): Observable<Marios> {
+    return this.http
+      .get<Marios>(`${this.baseUrl}/${mariosId}`)
+      .pipe(takeUntil(this.destroy$));
+  }
+
+  addMarios(payload: Marios): void {
+    this.http
+      .post<Marios>(this.baseUrl, payload)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((data) => {
+        this.mariosData = [...this.mariosData, data];
+        this.mariosList$.next(this.mariosData);
+      });
+  }
+
+  updateMarios(marios: Marios): void {
+    this.http
+      .put<Marios>(`${this.baseUrl}/${marios.id}`, marios)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((data) => {
+        this.mariosData = this.mariosData.map((m) =>
```

Comment:
m is not very descriptive name

Reply:
Fixed: 897376c596dd37aac60fd872a7d0d12a26bddf5b